lib/ClangImporter/ClangImporter.cpp

```diff
@@ -1022,6 +1022,16 @@ bool ClangImporter::isModuleImported(const clang::Module *M) {
   return M->NameVisibility == clang::Module::NameVisibilityKind::AllVisible;
 }
 
+bool ClangImporter::canImportModule(std::pair<Identifier, SourceLoc> moduleID) {
+  // Look up the top-level module to see if it exists.
+  // FIXME: This only works with top-level modules.  To support submodules, this
+  // should be replaced by checking the result of `loadModule`.
+  auto &clangHeaderSearch = Impl.getClangPreprocessor().getHeaderSearchInfo();
+  clang::Module *clangModule =
+    clangHeaderSearch.lookupModule(moduleID.first.str());
```

Comment:
Still looking for confirmation from @benlangmuir that this is an acceptable way to check for the existence of a module without loading it.